certbot/certbot/plugins/dns_common.py

```diff
@@ -197,7 +213,7 @@ def _prompt_for_data(label):
         :rtype: str
         """
 
-        def __validator(i): # pylint: disable=unused-private-member
+        def __validator(i: Optional[str]) -> None:  # pylint: disable=unused-private-member
```

Comment:
I have been mislead by the internal implementation of the `__validator` function, but indeed the outer context ensure the string is never `None`, but maybe be falsely because empty. I fixed the types.